api/sharedfolders/update

```diff
@@ -44,6 +44,14 @@ if($input->{'action'} eq 'edit') {
     $eventName = 'ibay-reset-permissions';
 } elsif($input->{'action'} eq 'migrate') {
     $eventName = 'vhost-migrate-ibay';
+} elsif ($input->{'action'} eq 'toggle-lock') {
```

Comment:
No need to duplicate the "toggle" logic in the API. The API needs just a "set-status" action with one boolean argument or just reuse the existing "edit" one.

Reply:
We use it also in the user account and in the virtualhosts UI, do you think it is a blocking point @DavidePrincipi

Reply:
Nevermind, I hope you understand my point though

Reply:
I like Setph the approach to separate the API with another action: we used the same pattern for all other APIs, one action, one job :)

Reply:
Indeed that's not the point. "Toggle" is a UI behavior and it makes no sense to implement it at the API level too. 
- The API code reads "status" from the DB to decide what to do. 
- The API returns "status" to the UI to let it decide what to show.

IMO the UI should pass the new "status" value to the API and this one has to apply that value.  The line R49 is useless.

I hope you see the point.